app/src/main/java/com/example/instagramclone/fragments/main_screen/compose/MainTopBar.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.instagramclone.fragments.main_screen.compose
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import com.example.instagramclone.R.drawable
+
+@Composable
+fun MainTopBar(onMessagesClick: () -> Unit) {
+    TopAppBar(
+        title = {
+            Image(
+                modifier = Modifier.size(108.dp),
+                painter = painterResource(id = drawable.ic_instagram_logo),
+                contentDescription = "InstagramLogo"
+            )
+        },
+        actions = {
+            MenuItems(onMessagesClick = { onMessagesClick() })
+        },
+        backgroundColor = Color.White,
+        elevation = 1.dp
+    )
+}
+
+@Composable
+fun MenuItems(onMessagesClick:() -> Unit) {
+    Image(
+        modifier = Modifier.size(22.dp),
+        painter = painterResource(id = drawable.ic_add_post_false),
+        contentDescription = "addpost"
+    )
+    Spacer(modifier = Modifier.width(25.dp))
+    Image(
+        modifier = Modifier
+            .size(22.dp)
+            .clickable {
+                onMessagesClick()
+            },
+        painter = painterResource(id = drawable.ic_messenger),
+        contentDescription = "messenger"
+    )
+    Spacer(modifier = Modifier.width(10.dp))
```

Comment:
Let's extract strings into resources ([reference](https://developer.android.com/jetpack/compose/resources#strings))

And it's best to use the [4-point grid system](https://medium.com/@aratidube12lns/4-point-grid-system-for-more-consistent-interface-design-efea81dea3f3#:~:text=What%20is%20the%204%2Dpoint,padding%2C%20margins%20and%20line%20heights.) wherever possible.